### PR TITLE
🐛 fix group, version, kind not being passed into util.ClusterToObjectsMapper

### LINKED ISSUE #842: Fix and verify unpaused Cluster triggers reconcile for dependent resources
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

The way unpause notifications have been implemented is incorrect and will not capture cluster unpause events. See #841. We should correct the rest of the unpause Cluster watches and author tests to verify they work.

### PR DESCRIPTION
**Pull request by devigned**
**What this PR does / why we need it**:
The way unpause notifications have been implemented incorrectly and will not capture cluster unpause events. See #841. This PR should correct the rest of the unpause Cluster watches

Todo:
- [x] add tests to verify behavior

**Which issue(s) this PR fixes**:
Fixes #842 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
fix group, version, kind not being passed into util.ClusterToObjectsMapper to trigger reconciles when Clusters are unpaused
```

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
As an extra validation, on top of controller unit tests, you might want to try enabling https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/test/e2e/capi_test.go#L94

It was failing on my first attempt to enable CAPI tests and this might very well be why

**Comment by CecileRobertMichon** [comment]
Have you tried creating a cluster manually and doing a clusterctl move and then seeing that reconciliation is actually unpaused? Not sure if the capi test is failing because it still doesn't work or if it's faulty

File: test/e2e/capi_test.go [hunk 1]
```diff
@@ -91,16 +91,15 @@ var _ = Describe("Running the Cluster API E2E tests", func() {
...
-	// 		return capi_e2e.SelfHostedSpecInput{
-	// 			E2EConfig:             e2eConfig,
-	// 			ClusterctlConfigPath:  clusterctlConfigPath,
-	// 			BootstrapClusterProxy: bootstrapClusterProxy,
-	// 			ArtifactFolder:        artifactFolder,
-	// 			SkipCleanup:           skipCleanup,
-	// 		}
-	// 	})
-	// })
+	Context("Running the self-hosted spec", func() {
```
Comment (CecileRobertMichon) [inline test/e2e/capi_test.go:94]: /hold

Let's make sure we don't merge this PR until we have a passing signal on the capi e2e test or undo this change otherwise it will break the periodic job

Comment (devigned) [reply test/e2e/capi_test.go:94]: For sure! Hopefully, it passes and we have one more test enabled.

File: cloud/services/networkinterfaces/networkinterfaces_test.go [hunk 2]
```diff
@@ -140,7 +141,7 @@ func TestReconcileNetworkInterface(t *testing.T) {
 				gomock.InOrder(
 					mSubnet.Get(context.TODO(), "my-rg", "my-vnet", "my-subnet").Return(network.Subnet{}, nil),
 					mLoadBalancer.Get(context.TODO(), "my-rg", "my-public-lb").Return(getFakeNodeOutboundLoadBalancer(), nil),
-					m.CreateOrUpdate(context.TODO(), "my-rg", "my-net-interface", matchers.DiffEq(network.Interface{
+					m.CreateOrUpdate(context.TODO(), "my-rg", "my-net-interface", gomockinternal.DiffEq(network.Interface{
```
Comment (CecileRobertMichon) [inline cloud/services/networkinterfaces/networkinterfaces_test.go:144]: is it better to use gomockinternal? Asking because I've been using matchers.DiffEq everywhere so wondering if I should stop doing that

Comment (devigned) [reply cloud/services/networkinterfaces/networkinterfaces_test.go:144]: The are the same thing. I moved an the `internal/matchers` package to `internal/matchers/gomock`, b/c I needed to make matchers for gomega too.

**Comment by devigned** [comment]
@CecileRobertMichon looks like pull-cluster-api-provider-azure-capi-e2e didn't pass. I'm going to dig into it a bit, but if it's not initially clear, I'm leaning toward commenting and taking it on in a new issue. Thoughts?

**Comment by CecileRobertMichon** [comment]
@devigned sgtm, if that's the case let's tackle it as part of #780 
cc @nader-ziada

**Comment by devigned** [comment]
I'm running the test locally. I'll have a definitive answer about what's going on shortly.

**Comment by devigned** [comment]
@CecileRobertMichon I've commented out the CAPI Self-Host test again and opened #852 to address the issues causing the test to fail. I think this PR is ready for final review after passing tests.

**Comment by devigned** [comment]
/hold
flaky test

**Comment by devigned** [comment]
I think I've resolved the flaky test issue. If `azure-test` passes without complaining about logs, I think the change set is ready to go.

**Comment by devigned** [comment]
> the exp package common tests should be the same as the non-exp ones, just need to test in two places for now but we should be able to get rid of them once these features move out of exp/ correct?

Correct. We should remove the exp test and add the controller names into the non-exp tests when out of "experimental".

> why delete the tests in azurecluster_controller_test but leave the ones in azuremachine_controller_test since it looks like the common tests cover both?

Good catch. That's extra code. I'll remove it.
